osu.Game/Overlays/Dashboard/Friends/FriendsLayout.cs

```diff
@@ -0,0 +1,252 @@
+                                        AutoSizeAxes = Axes.Y,
+                                        Padding = new MarginPadding
+                                        {
+                                            Horizontal = 40,
+                                            Vertical = 20
+                                        },
+                                        Child = userListToolbar = new UserListToolbar
+                                        {
+                                            Anchor = Anchor.CentreRight,
+                                            Origin = Anchor.CentreRight,
+                                        }
+                                    },
+                                    new Container
+                                    {
+                                        RelativeSizeAxes = Axes.X,
+                                        AutoSizeAxes = Axes.Y,
+                                        Children = new Drawable[]
+                                        {
+                                            itemsPlaceholder = new Container
+                                            {
+                                                RelativeSizeAxes = Axes.X,
+                                                AutoSizeAxes = Axes.Y,
+                                                Padding = new MarginPadding { Horizontal = 50 }
+                                            },
+                                            loading = new LoadingLayer(itemsPlaceholder)
+                                        }
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+            };
+        }
+
+        [BackgroundDependencyLoader]
+        private void load(OverlayColourProvider colourProvider)
+        {
+            background.Colour = colourProvider.Background4;
+            controlBackground.Colour = colourProvider.Background5;
+        }
+
+        protected override void LoadComplete()
+        {
+            base.LoadComplete();
+
+            onlineStatusControl.Current.BindValueChanged(_ => recreatePanels());
+            userListToolbar.DisplayStyle.BindValueChanged(_ => recreatePanels());
+            userListToolbar.SortCriteria.BindValueChanged(_ => recreatePanels());
+
+            if (!api.IsLoggedIn)
+                return;
+
+            request = new GetFriendsRequest();
+            request.Success += response => Schedule(() => Users = response);
+            api.Queue(request);
+        }
+
+        private void recreatePanels()
+        {
+            if (!users.Any())
+                return;
+
+            cancellationToken?.Cancel();
+
+            if (itemsPlaceholder.Any())
+                loading.Show();
+
+            var groupedUsers = onlineStatusControl.Current.Value?.Users ?? new List<User>();
```

Comment:
Needs a better name like `usersInCurrentGroup`. I got confused at what it is for a second.